src/features/bulk/bulk/sagas.ts

```diff
@@ -0,0 +1,71 @@
+import { select, takeEvery } from 'redux-saga/effects';
+import { resetState, stepForward, stepBack } from './slice';
+import { BulkStepsToRoutes } from './constants';
+import { getCurrentStep, getFiles } from './selectors';
+import { CurrentStep } from './model';
+import { CustomFileLight } from '../../../types';
+import { NavigateFunction } from 'react-router-dom';
+
+interface ActionType {
+	type: string;
+	payload: {
+		navigate: NavigateFunction;
+	};
+}
+
+function* back(action: ActionType) {
+	try {
+		const currentStep: CurrentStep = yield select(getCurrentStep);
+		const { navigate } = action.payload;
+
+		switch (currentStep) {
+			case CurrentStep.EditFields:
+				navigate(BulkStepsToRoutes[CurrentStep.SelectFields]);
+				break;
+			case CurrentStep.SelectFields:
+				navigate(BulkStepsToRoutes[CurrentStep.Upload]);
+				break;
+			case CurrentStep.Upload:
+				navigate(BulkStepsToRoutes[CurrentStep.Button]);
+				break;
+		}
+	} catch (e: any) {
+		console.log('back fail', e);
```

Comment:
Catching an error and then only logging it is not very useful I'd argue. I actually don't understand what would trigger the catch looking at this. I'm sure you can do better than typing the exception as any as well :slightly_smiling_face:

Reply:
The old code some @todos for errors as well. What would be the prefered way to handle these errors?